hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/BaseSparkInternalRowReaderContext.java

```diff
@@ -91,4 +93,15 @@ protected UnaryOperator<InternalRow> getBootstrapProjection(Schema from, Schema
         HoodieInternalRowUtils.getCachedUnsafeRowWriter(getCachedSchema(from), getCachedSchema(to), Collections.emptyMap(), partitionValuesByIndex);
     return row -> (InternalRow) unsafeRowWriter.apply(row);
   }
+
+  @Override
+  public void setSchemaHandler(FileGroupReaderSchemaHandler<InternalRow> schemaHandler) {
+    super.setSchemaHandler(schemaHandler);
+    // init ordering value converter: java -> engine type
+    List<String> orderingFieldNames = HoodieRecordUtils.getOrderingFieldNames(getMergeMode(), tableConfig);
+    Schema schema = schemaHandler.getRequiredSchema();
+    if (orderingFieldNames.stream().allMatch(f -> AvroSchemaUtils.findNestedField(schema, f).isPresent())) {
```

Comment:
is the check necessary because there is already existence check in the `OrderingValueEngineTypeConverter `